fauna/client/client.py

```diff
@@ -579,6 +617,39 @@ def _set_endpoint(self, endpoint):
     self._endpoint = endpoint
 
 
+class RetryStreamIter:
+  """A frankenstein class that mix a ContextManager and an Iterator so we can detected retryable errors."""
+
+  def __init__(self, client, fql):
+    self.client = client
+    self.fql = fql
+    self.stream = self.client._stream(self.fql)
+    self.response = None
+
+  def __enter__(self):
+    self.response = self.stream.__enter__()
+    return self
+
+  def __exit__(self, exc_type, exc_value, exc_traceback):
+    self.stream.__exit__(exc_type, exc_value, exc_traceback)
+    return None
+
+  def __iter__(self):
+    return self
+
+  def __next__(self):
+    try:
+      return next(self.response)
+    except (RetryableFaunaException, httpx.TimeoutException, httpx.NetworkError,
+            httpx.ProtocolError) as ex:
+      self.stream = self.client._stream(self.fql)
```

Comment:
We gotta keep track of the last seen event ts and send that back to the server during retries. Otherwise the stream starts over from the beginning.

Reply:
the approach I was taking here is to reissue a query to grab a new stream token, looks like I can avoid this just by keep track of the last seen ts (and use the initial stream token)?

Reply:
That's right! 

The current approach would lose events in between the error and the time to reconnect instead of replaying repeated events. If you track the event ts, you can avoid both issues.